### PR TITLE
Fix documentation build (+ tweaks)

### LINKED ISSUE #422: Documentation broken
I am working on it. Just opening it here so that no else wastes his time on this.

### PR DESCRIPTION
**Pull request by thofma**
- Bump Hecke version to suppress warnings
- Make (random) doctest a `julia` block
- Remove illegal bibtex entries
- Add dependencies to build AA documentation
- Remove duplicated docstring for `quo` from `Rings/affine.md`
- `checkdocs = :none` to suppress the report of the non-documented functions
- `strict = true` to catch regressions earlier

Fixes #422.

### CODE REVIEW

File: experimental/ModStd/ModStdQt.jl [hunk 1]
```diff
@@ -517,11 +517,14 @@ The return value is an array
...
-julia>  f = (X[1]^2+a[1]*X[2]^2)*(X[1]+X[2]);
-julia>  z = factor_absolute(f)
+```julia
+julia> Qa, a = PolynomialRing(QQ, :a=>1:2);
+
+julia> R, X = PolynomialRing(FractionField(Qa), :X=>1:2);
+
+julia> f = (X[1]^2+a[1]*X[2]^2)*(X[1]+X[2]);
+
+julia> z = factor_absolute(f)
```
Comment (tthsqe12) [inline experimental/ModStd/ModStdQt.jl:528]: I do not thing this PR fixes the abs factor: some factors can be returned in the wrong parent. Maybe the example should be changed to this and the code fixed?
```
julia> Qa, a = PolynomialRing(QQ, :a=>1:2);

julia> R, X = PolynomialRing(FractionField(Qa), :X=>1:2);

julia> f = (X[1]^2+a[1]*X[2]^2)*(X[1]+X[2]+a[1]+a[2]);

julia> z = factor_absolute(f)
3-element Vector{Any}:
 1
 (x1 + x2 + x3 + x4, 1)
 (X1 + t*X2, X1 - t*X2, 1)

julia> parent(z[2][1])
Multivariate Polynomial Ring in x1, x2, x3, x4 over Rational Field
```

Comment (thofma) [reply experimental/ModStd/ModStdQt.jl:528]: Fixed the example.

**Comment by tthsqe12** [comment]
I think you misunderstood my comment. The function should not be returning a factor in the temp parent QQ[x1,x2,x3,x4]. The factor (X1+X2+a1+a2) should come back in Q(a1,a2)[X1,X2]. There is a problem with the factor function.

**Comment by thofma** [comment]
You should comment this in #420.

I just want the documentation to build.

**Comment by tthsqe12** [comment]
ok. sorry.

**Pull request merged**